io/net/virt-net/lpm.py

```diff
@@ -0,0 +1,329 @@
+
+        pxh.login(ipaddr, username, password)
+        pxh.sendline()
+        pxh.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        pxh.set_unique_prompt()
+        pxh.prompt(timeout=60)
+        self.pxssh = pxh
+
+    def run_command(self, command, timeout=3000):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on hmc: %s", command)
+        con = self.pxssh
+        con.sendline(command)
+        con.expect("\n")  # from us
+        con.expect(con.PROMPT, timeout=timeout)
+        output = con.before.splitlines()
+        con.sendline("echo $?")
+        con.prompt(timeout)
+        return output
+
+    def rsct_service_start(self):
+        '''
+        Running rsct services which is necessary for Network
+        virtualization tests
+        '''
+        try:
+            for svc in ["rsct", "rsct_rm"]:
+                process.run('startsrc -g %s' % svc, shell=True, sudo=True)
+        except CmdError as details:
+            self.log.debug(str(details))
+            self.fail("Starting service %s failed", svc)
+
+        output = process.system_output("lssrc -a", ignore_status=True,
+                                       shell=True, sudo=True)
+        if "inoperative" in output:
+            self.cancel("Failed to start the rsct and rsct_rm services")
+
+    def install_packages(self):
+        '''
+        Install required packages
+        '''
+        smm = SoftwareManager()
+        packages = ['ksh', 'src', 'rsct.basic', 'rsct.core.utils',
+                    'rsct.core', 'DynamicRM']
+        detected_distro = distro.detect()
+        self.log.info("Test is running on: %s", detected_distro.name)
+        for pkg in packages:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel('%s is needed for the test to be run' % pkg)
+
+    def set_msp(self, server, vios_id):
+        '''
+        Sets the msp for the specified VIOS
+        '''
+        cmd = "chsyscfg -m %s -r lpar -i lpar_id=%s,msp=1" % (server,
+                                                              vios_id)
+        self.run_command(cmd)
+
+    def do_migrate(self, server, remote_server, lpar, params):
+        '''
+        Migrate the LPAR from server to remote server, with additional
+        params specified.
+        '''
+        cmd = "migrlpar -o m -m %s -t %s -p %s %s" % (server,
+                                                      remote_server,
+                                                      lpar, params)
+        self.log.debug("\n".join(self.run_command(cmd)))
+        time.sleep(10)
+        if not self.is_lpar_in_server(remote_server, lpar):
+            self.fail("%s not in %s" % (lpar, remote_server))
+        time.sleep(300)
```

Comment:
Can the sleep time be made configurable? Not sure if different network will have different impact?

Reply:
I am yet to find a suitable way to make sure it is stable. Until then, sleep is needed.
I will make it configurable, and add a todo also

Reply:
Yeah, I meant `yaml` configurable.

Reply:
@harish-24 Yes, I understood you meant `yaml` configurable. I do not want to give the user that option since user is not the best person to decide that. I am checking to find an way to decide on system stability after LPM. So, have added a todo. When I am not able to find out the way, I will add this as yaml configurable, as already mentioned.

Is this fine with you ?

Reply:
ok